src/main/resources/db/migration/V0002__purge.sql

```diff
@@ -0,0 +1,259 @@
+                SELECT service_request_id
+                FROM sdt_owner.service_requests
+                WHERE request_timestamp < t_retention_date));
+
+-- cursor4 error logs for individual requests for bulk submissions
+--   with service requests created before given retention date
+c4 CURSOR
+    FOR SELECT error_log_id
+        FROM sdt_owner.error_logs
+        WHERE individual_request_id IN (
+            SELECT individual_request_id
+            FROM sdt_owner.individual_requests
+            WHERE bulk_submission_id IN (
+                SELECT bulk_submission_id
+                FROM sdt_owner.bulk_submissions
+                WHERE service_request_id IN (
+                    SELECT service_request_id
+                    FROM sdt_owner.service_requests
+                    WHERE request_timestamp < t_retention_date)));
+
+n_iteration   INT;
+
+BEGIN
+
+    -- retention period
+	SELECT parameter_value
+    INTO i_retention_period
+    FROM sdt_owner.global_parameters
+    WHERE parameter_name = 'DATA_RETENTION_PERIOD';
+
+    -- job start date
+	SELECT now()
+    INTO t_job_start_date;
+    RAISE NOTICE 'Job start date: %', t_job_start_date;
+
+    -- retention_date calculated using retention period and now
+	SELECT t_job_start_date::DATE - (i_retention_period || ' DAY')::INTERVAL
+    INTO t_retention_date;
+    RAISE NOTICE 'Retention period date: %', t_retention_date;
+
+	-- get job audit id
+	SELECT nextval('purge_job_audit_seq')
+	INTO i_purge_job_id;
+
+    RAISE NOTICE 'i_purge_job_id is %', i_purge_job_id;
+
+    -- create job audit
+	INSERT INTO sdt_owner.purge_job_audit
+  	(purge_job_id, job_start_date, retention_date, success)
+  	VALUES (i_purge_job_id, t_job_start_date, t_retention_date, 0);
+
+    RAISE NOTICE 'Purge retention period is % days, deleting all data prior to %', i_retention_period, t_retention_date;
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+			'Purge retention period is ' || i_retention_period ||
+			' days, deleting all data prior to ' || t_retention_date);
+
+    -- delete error logs
+    RAISE NOTICE 'Commencing deletion of error_logs records';
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+			'Commencing deletion of error_logs records');
+
+	n_iteration := 0;
+    FOR c4_rec IN c4 LOOP
+        DELETE FROM sdt_owner.error_logs
+        WHERE error_log_id = c4_rec.error_log_id;
+
+	    RAISE NOTICE 'Deleted error_logs : %', c4_rec.error_log_id;
+
+	    INSERT INTO sdt_owner.purge_job_audit_messages
+		(purge_job_message_id, purge_job_id, message_date,log_message)
+    	VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+				'Deleted error_logs : ' || c4_rec.error_log_id);
```

Comment:
The original version of the procedure didn't log the deletion of individual error logs.  Any reason it's been added in this version?

Reply:
Gone